tests/array_split.py

```diff
@@ -22,127 +22,110 @@
+            if even_div != uneven_div and uneven_div is not None:
+                input_arr.append(uneven_div)
+            # integer larger than shape[axis]
+            input_arr.append(a.shape[axis] + np.random.randint(1, 10))
+            # indices array which has points
+            # within the target dimension of the src array
+            if a.shape[axis] > 1:
+                input_arr.append(list(range(1, a.shape[axis], even_div)))
+            # indices array which has points
+            # out of the target dimension of the src array
+            input_arr.append(
+                list(
+                    range(
+                        0,
+                        a.shape[axis] + even_div * np.random.randint(1, 10),
+                        even_div,
                     )
-                    input_arr.append(
-                        list(
-                            range(
-                                a.shape[axis]
-                                + even_div * np.random.randint(1, 10),
-                                0,
-                                -even_div,
-                            )
-                        )
+                )
+            )
+            input_arr.append(
+                list(
+                    range(
+                        a.shape[axis] + even_div * np.random.randint(1, 10),
+                        0,
+                        -even_div,
                     )
+                )
+            )
 
-                    for routine, input_opt in itertools.product(
-                        test_routine, input_arr
-                    ):
-                        # test divisible integer or indices
-                        print_msg = (
-                            f"np.{routine}({a.shape}, {input_opt}" f", {axis})"
-                        )
-                        # Check if both impls produce the error
-                        # for non-viable options
-                        b = getattr(np, routine)(a, input_opt, axis)
-                        c = getattr(num, routine)(a, input_opt, axis)
-                        is_equal = True
-                        err_arr = [b, c]
+            for routine, input_opt in itertools.product(
+                test_routine, input_arr
+            ):
+                # test divisible integer or indices
+                print_msg = f"np.{routine}({a.shape}, {input_opt}" f", {axis})"
+                # Check if both impls produce the error
+                # for non-viable options
+                b = getattr(np, routine)(a, input_opt, axis)
+                c = getattr(num, routine)(a, input_opt, axis)
+                is_equal = True
+                err_arr = [b, c]
 
-                        if len(b) != len(c):
+                if len(b) != len(c):
+                    is_equal = False
+                    err_arr = [b, c]
+                else:
+                    for each in zip(b, c):
+                        sub_set = list(each)
```

Comment:
why do you need to create a fresh list here? isn't each already a tuple that you can access with index expression?